### PR TITLE
fixes #24 to make it easier to extend the available commands

### LINKED ISSUE #24: make it easier to extend to add other forge commands
right now commands are totally hard coded due to the regex + messages. Then the use of `@Pattern` only allows real constants it seems .

So here's a PR that validates commands are in the allowed commandMap and generates a dynamic error message based on the loaded commands (so that we can more easily configure/extend which commands are supported).

### PR DESCRIPTION
**Pull request by jstrachan**

### CODE REVIEW

File: src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java [hunk 1]
```diff
@@ -70,14 +69,26 @@
...
-{
-   private static final String ALLOWED_CMDS_VALIDATION_MESSAGE = "Supported commmands are 'obsidian-new-quickstart' or 'obsidian-new-project'";
-   private static final String ALLOWED_CMDS_PATTERN = "(obsidian-new-quickstart)|(obsidian-new-project)";
+public class ObsidianResource {
    private static final String DEFAULT_COMMAND_NAME = "obsidian-new-quickstart";
 
    private static final Logger log = Logger.getLogger(ObsidianResource.class.getName());
-   private Map<String, String> commandMap = new HashMap<>();
+
+   private static String join(Iterable<String> strings, String separator) {
```
Comment (gastaldi) [inline src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java]: Can we use JDK8's `String.join` instead?

File: src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java [hunk 2]
```diff
@@ -70,14 +69,26 @@
...
+            first = false;
+         } else {
+            buffer.append(separator);
+         }
+         buffer.append(string);
+      }
+      return buffer.toString();
+   }
+
+   private final Map<String, String> commandMap = new TreeMap<>();
```
Comment (gastaldi) [inline src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java]: I think a Set would be better in here

File: src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java (hunk 1 above)
Comment (jstrachan) [reply src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java]: happy to yeah - kept it safe wasn't sure you were cool yet with JDK 8 only

File: src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java (hunk 2 above)
Comment (jstrachan) [reply src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java]: seems the code uses the Map though right? It looks up the value in the map

Comment (jstrachan) [reply src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java]: see: https://github.com/obsidian-toaster/generator-backend/blob/master/src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java#L296

Comment (gastaldi) [reply src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java]: Ah right, sorry I missed that

**Comment by gastaldi** [comment]
Thanks!

**Pull request merged**